src/Avalonia.Base/Rendering/Composition/Server/ServerCompositor.cs

```diff
@@ -168,8 +165,14 @@ void NotifyBatchesRendered()
             _reusableToNotifyRenderedList.Clear();
         }
 
-        public void Render() => Render(true);
-        public void Render(bool catchExceptions)
+        // This Render overload can be called outside of the compositor, like MediaContext.SyncCommit.
+        // In this case we can't use external timestamp, as it might be out-sync with IRenderLoop.
+        // Instead, pass last known ServerNow time.
+        public void Render(bool catchExceptions) => Render(ServerNow, catchExceptions);
```

Comment:
Since that can potentially trigger issues with server animations on window resize, maybe we should add a way to pass a time provider to Compositor on creation? That should cover headless platform and unit testing needs.

We can rethink the render loop time thingy once one of our render loops becomes aware of actual presentation times, though it's more likely for the compositor itself to know that.

Reply:
The problem is that I didn't introduce any time provider (which lives in a separated polyfill package) dependencies here.

Reply:
Done